lib/jsonapi_compliable.rb

```diff
@@ -1,20 +1,75 @@
+require 'json'
+require 'active_support/core_ext/string'
+require 'active_support/core_ext/class/attribute'
+require 'active_support/concern'
+
+require 'jsonapi/serializable'
+# Temporary fix until fixed upstream
+# https://github.com/jsonapi-rb/jsonapi-serializable/pull/102
+JSONAPI::Serializable::Resource.class_eval do
+  def requested_relationships(fields)
+    @_relationships
+  end
+end
+
+# This library looks up a serializer based on the record's class name
+# This wouldn't work for us, since a model may be associated with
+# multiple resources.
+# Instead, this variable is assigned when the query is resolved
+# To ensure we always render with the *resource* serializer
+JSONAPI::Serializable::Renderer.class_eval do
```

Comment:
Have you considered [`Module#prepend`](https://medium.com/@sebastianroyer/prepend-and-super-for-elegant-monkey-patching-24a5b98f4468) as an alternative to monkey patching?

Reply:
I think either way it's a monkey patch; seems 6 in one, half dozen in the other. The real solution will be to stabilize and release these changes then hook up with @beauby to see what we can do upstream.

Reply:
It's hard to track down the source of the monkey patch; but the Prepended module give the reader an easy indication. Example:

```ruby
class Foo
end

Foo.class_eval do
  def monkey_patched
  end
end

module JSONAPIPatch
  def patched
  end
end

Foo.prepend(JSONAPIPatch)
```
```pry
[1] pry(#<Foo>)> ls
Foo#methods: monkey_patched
JSONAPIPatch#methods: patched
```

There are many other drawbacks to monkey patching, and they're well documented; so won't repeat them here.